pymodm/fields.py

```diff
@@ -1172,24 +1198,30 @@ def contribute_to_class(self, cls, name):
                                                     name,
                                                     self._on_delete)
 
-    def to_python(self, value):
-        if isinstance(value, dict):
-            # Try to convert the value into our document type.
-            try:
-                return self.related_model.from_document(value)
-            except (ValueError, TypeError):
-                pass
-
+    def dereference_if_needed(self, value):
         if isinstance(value, self.related_model):
             return value
-
         if self.model._mongometa._auto_dereference:
             # Attempt to dereference the value as an id.
             dereference_id = _import('pymodm.dereference.dereference_id')
             return dereference_id(self.related_model, value)
-
         return self.related_model._mongometa.pk.to_python(value)
 
+    def to_python(self, value):
+        # Default/blank values can be returned immediately.
+        if self.is_blank(value):
+            return value
+
+        # Attempt casting to referenced model.
+        if isinstance(value, dict):
+            try:
+                return self.related_model.from_document(value)
+            except (ValueError, TypeError):
+                pass
```

Comment:
Unrelated to this review but there's a bug in this logic when using dicts as _id. This is pretty obscure so I'm not sure it's worth trying to fix. 
```python
    def test_dereference_dict_id(self):

        class OtherModel(MongoModel):
            _id = fields.OrderedDictField(primary_key=True)
            a = fields.IntegerField()
            b = fields.IntegerField()

        class Container(MongoModel):
            ref = fields.ReferenceField(OtherModel)

        from collections import OrderedDict
        m1 = OtherModel(OrderedDict([('a', 1), ('b', 2)]), 3, 4).save()
        m2 = Container(m1.pk)
        self.assertEqual(m2.ref, m1)
```

Reply:
Created [PYMODM-80](https://jira.mongodb.org/browse/PYMODM-80)